src/util/convertHTMLtoJSON.ts

```diff
@@ -0,0 +1,178 @@
+import * as htmlparser from 'htmlparser2'
+import _ from 'lodash'
+import { Context, Path } from '../types'
+import { State } from './initialState'
+
+/** Returns true if the given tagname is ul or ol. */
+const isList = (tagname: string) => tagname === 'ul' || tagname === 'ol'
+
+/** Returns true if the given tagname is li or p. */
+const isListItem = (tagname: string) => tagname === 'li' || tagname === 'p'
+
+/** Returns true if the given tagname is i, b, or u. */
+const isFormattingTag = (tagname: string) => tagname === 'i' || tagname === 'b' || tagname === 'u'
+
+interface ImportHtmlOptions {
+  skipRoot? : boolean,
+}
+interface InsertThoughtOptions {
+  indent?: boolean,
+  outdent?: boolean,
+  insertEmpty?: boolean,
+}
+interface ThoughtJSON {
+  scope: string,
+  children: ThoughtJSON[],
+}
```

Comment:
Let's use the `Block` type that is already declared in `importText.ts`. I'm not sure if it ends up being used in more than one file, but if so you can add it to the general `types.ts`. Otherwise define it only in the file that uses it.